### DISCUSSION TITLE
Construct a `nb::callable` from a `nb::object`

### DISCUSSION THREAD

**Discussion by alkino** [wjakob/nanobind#671] [General]
So I have a `nb::object` and I know this is a callable, I want to do a `nb::callable` with that to be more clear.

```
nb::object po = function_foo_bar();
nb::callable callable(po);
```

`g++` returns: `an inherited constructor is not a candidate for initialization from an expression of the same or derived type`

And this is true the correct constructor here that is `nb::callable(nb::object& obj)` came from `using object::object`.

I finally found by reading how accessors where working that I can do:
`nb::callable call(nb::borrow<nb::callable>(po));`

Is it ok? Is there a better way? Is it expected?

**Comment by wjakob** [comment]
The `nb::borrow<nb::callable>(po)` pattern (although not very pretty) is indeed what I would use.
